e2e/cache/cache.go

```diff
@@ -18,88 +20,258 @@ type cacheTests struct {
+}
+
+func (c *cacheTests) testNoninteractiveCacheCmds(t *testing.T) {
+	tests := []struct {
+		name               string
+		options            []string
+		needImage          bool
+		expectedEmptyCache bool
+		expectedOutput     string
+		exit               int
+	}{
+		{
+			name:               "clean force",
+			options:            []string{"clean", "--force"},
+			expectedOutput:     "",
+			needImage:          true,
+			expectedEmptyCache: true,
+			exit:               0,
+		},
+		{
+			name:           "clean help",
+			options:        []string{"clean", "--help"},
+			expectedOutput: "Clean your local Singularity cache",
+			needImage:      false,
+			exit:           0,
+		},
+		{
+			name:           "list help",
+			options:        []string{"list", "--help"},
+			expectedOutput: "List your local Singularity cache",
+			needImage:      false,
+			exit:           0,
+		},
+		{
+			name:               "list type",
+			options:            []string{"list", "--type", "library"},
+			needImage:          true,
+			expectedOutput:     "There are 1 container file",
+			expectedEmptyCache: false,
+			exit:               0,
+		},
+		{
+			name:               "list verbose",
+			needImage:          true,
+			options:            []string{"list", "--verbose"},
+			expectedOutput:     "NAME",
+			expectedEmptyCache: false,
+			exit:               0,
+		},
+	}
+
+	// A directory where we store the image and used by separate commands
+	tempDir, imgStoreCleanup := e2e.MakeTempDir(t, "", "", "image store")
+	defer imgStoreCleanup(t)
+	imagePath := filepath.Join(tempDir, imgName)
+
+	for _, tt := range tests {
+		// Each test get its own clean cache directory
+		cacheDir, cleanup := e2e.MakeCacheDir(t, "")
+		defer cleanup(t)
+		h, err := cache.NewHandle(cacheDir)
+		if err != nil {
+			t.Fatalf("Could not create image cache handle: %v", err)
 		}
 
-		c := &cacheTests{
-			env: env,
+		if tt.needImage {
+			prepTest(t, c.env, tt.name, h, imagePath)
+		}
+
+		c.env.ImgCacheDir = cacheDir
+		c.env.RunSingularity(
+			t,
+			e2e.AsSubtest(tt.name),
+			e2e.WithCommand("cache"),
+			e2e.WithArgs(tt.options...),
+			e2e.ExpectExit(tt.exit),
+		)
+
+		if tt.needImage && tt.expectedEmptyCache {
+			ensureCacheEmpty(t, tt.name, imagePath, h)
 		}
-		t.Run("cacheClean", c.testCacheClean)
 	}
 }
 
-func (c *cacheTests) testCacheClean(t *testing.T) {
+func (c *cacheTests) testInteractiveCacheCmds(t *testing.T) {
 	tt := []struct {
-		name    string
-		options []string
-		expect  string
-		send    string
-		output  string
-		exit    int
+		name               string
+		options            []string
+		expect             string
+		send               string
+		exit               int
+		expectedEmptyCache bool // Is the cache supposed to be empty after the command is executed
 	}{
 		{
-			name:   "clean cache normal",
-			expect: "Do you want to continue? [N/y]",
-			send:   "y",
-			exit:   0,
+			name:               "clean normal",
```

Comment:
Do we want to test any of the `N` options here? Just to make sure they don't do anything when completed?

Reply:
I can easily add that, no problem.